include/deal.II/lac/full_matrix.templates.h

```diff
@@ -1165,6 +1165,46 @@ FullMatrix<number>::operator == (const FullMatrix<number> &M) const
 }
 
 
+namespace internal
+{
+  namespace
+  {
+    // LAPACKFullMatrix is not implemented for
+    // complex numbers or long doubles
+    template<typename number, typename = void>
+    struct Determinant
+    {
+      static number value (const FullMatrix<number> &A)
+      {
+        AssertThrow(false, ExcNotImplemented());
+        return 0.0;
+      }
+    };
+
+
+#ifdef DEAL_II_WITH_LAPACK
```

Comment:
I think I would prefer if you put the preprocessor `#ifdef` into the function body, with an `#else` that says something like that this function is not implemented if not using LAPACK. Right now, if you don't use LAPACK, the compiler will select the general template above, which gives no indication that the problem is that LAPACK wasn't found.

Maybe just use the same exception message that we use in all of the other places where we error out because LAPACK wasn't found.